src/tests/multihost/alltests/test_inmemory_cache.py

```diff
@@ -0,0 +1,329 @@
+        client = multihost.client[0]
+        client.run_command("sss_cache -U --domain=example1")
+        client.run_command("strace -o /tmp/straceuser3 getent passwd foo1")
+        find_logs(multihost, "/tmp/straceuser3", "/var/lib/sss/pipes/nss")
+        find_logs(multihost, "/tmp/straceuser3", "/var/lib/sss/mc/passwd")
+        find_logs(multihost, log_sssd, "Got request for ")
+        find_logs(multihost, log_sssd, "uid=foo1")
+        clear_only_domain_log(multihost)
+        client.run_command("sss_cache -G --domain=example1")
+        client.run_command("strace -o /tmp/stracegroup3 getent group ldapusers")
+        find_logs(multihost, "/tmp/stracegroup3", "/var/lib/sss/pipes/nss")
+        find_logs(multihost, "/tmp/stracegroup3", "/var/lib/sss/mc/group")
+        find_logs(multihost, log_sssd, "Got request for ")
+        find_logs(multihost, log_sssd, "cn=ldapusers")
+
+    @staticmethod
+    def test_bz867933(multihost):
+        """
+        :title: sss cache invalidates the InMemory cache when sssd is stopped bz867933
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=867933
+        :id: 0db6d2e0-96ee-11ed-bbb9-845cf3eff344
+        :steps:
+            1. Make the API to invalidate the cache public
+            2. In the sss_cache tool, carefully detect if the sss_nss process is running
+            3. if the sss_nss process is running, proceed with sending the signal
+            4. if the sss_nss process is not running, invalidate the memcache.
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+            4. Should succeed
+        """
+        client = multihost.client[0]
+        log_sssd = f'/var/log/sssd/sssd_{ds_instance_name}.log'
+        client.run_command("service sssd stop")
+        client.run_command("sss_cache -U --domain=example1")
+        client.run_command("sss_cache -G --domain=example1")
+        client.run_command("service sssd start")
+        clear_only_domain_log(multihost)
+        client.run_command("getent passwd foo1")
+        client.run_command("getent group ldapusers")
+        find_logs(multihost, log_sssd, "Got request for ")
+        find_logs(multihost, log_sssd, "uid=foo1")
+        find_logs(multihost, log_sssd, "cn=ldapusers")
+
+    @staticmethod
+    def test_memcache(multihost):
+        """
+        :title: Set value of option memcache timeout to 0 and expire InMemory cache immediately
+        :id: 19dfa7ae-96e9-11ed-adf3-845cf3eff344
+        :steps:
+            1. Set value of option memcache timeout to 0 and expire InMemory cache immediately
+            2. Validate the user using sss cache
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+        """
+        client = multihost.client[0]
+        tools = sssdTools(multihost.client[0])
+        tools.sssd_conf("nss", {'memcache_timeout': '0'}, action='update')
```

Comment:
You are changing the config, backupsssdconf fixture should be used to restore the config.